### PR TITLE
Consolidate requirements

### LINKED ISSUE #172: Requirements mismatch as defined in multiple locations
There seems to be a mismatch of requirements between `setup.py`, `requirements.txt` and `requirements/prod.txt`

eg numpy==1.16 vs 1.16.2

Is there a reason for the three different lists of requirements? Can we consolidate?

### PR DESCRIPTION
**Pull request by forslund**
This removes the requirements.txt and package list in setup.py in favor of the list in requirements/prod.txt and should resolve #172

It also bumps the numpy version to 1.16.2 to match the version listed in requirements.txt which should fix some errors with pyinstaller.

### CODE REVIEW

**Comment by jessecooper** [comment]
Any reason this is hanging around for over a year?

**Comment by NoamDev** [comment]
Maybe @MatthewScholefield  or @krisgesling can review this?

File: requirements.txt [hunk 1]
```diff
@@ -1,44 +0,0 @@
-absl-py==0.7.1
```
Comment (krisgesling) [inline requirements.txt:1]: What were the rest of these requirements for? 

Do we need a new `requirements/dev.txt` or something?

**Review by krisgesling — COMMENTED** [review]
Thanks for the nudge, this looks like a great change.

One question below

File: requirements/prod.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-numpy==1.16
+numpy==1.16.2
```
Comment (jessecooper) [inline requirements/prod.txt:1]: This can be bumped to 1.16.5

Comment (jessecooper) [reply requirements/prod.txt:1]: also `h5py<3.0.0` fixes a runtime problem

**Comment by ShamoX** [comment]
Hello,

Since I need that project, I decided to fork and advance. Sorry for the inconvenience but it can be reversed if Mycroft wants of course).

You will find your PR to the fork [here](https://gitlab.com/liant-sasu/mycroft-precise/-/merge_requests/8).

Again, sorry for the inconvenient. I created a [thread](https://community.mycroft.ai/t/mycroft-precise-delivery-hanging/12348) on Mycroft community forum to discuss if needed.

**Comment by ShamoX** [comment]
Be careful, those changes generate a regression: wheel cannot build anymore, since setup.py needs requirements/prod.txt to be part of the package and is not.

I revert the change and propose to remove entirely requirements/prod.txt and use correctly setup.py list: https://gitlab.com/liant-sasu/mycroft-precise/-/merge_requests/13

**Comment by jessecooper** [comment]
@ShamoX https://github.com/jessecooper/mycroft-precise/tree/dev/requirements
This is what I am currently running with.

**Pull request closed without merging**